File: 1768/python/Solution.py
class Solution:
    def mergeAlternately(self, word1: str, word2: str) -> str:
        A, B = len(word1), len(word2)
        a, b = 0, 0
        result = []

        while a < A and b < B:
            result.append(word1[a])
            result.append(word2[b])
            a += 1
            b += 1

        # Add remaining characters from either string
        result.extend(word1[a:])
        result.extend(word2[b:])

        return "".join(result)

File: 1768/csharp/Solution.cs
using System;

public class Solution {
    public string MergeAlternately(string word1, string word2) {
        int A = word1.Length, B = word2.Length;
        int a = 0, b = 0;
        List<string> s = new List<string>();
        int word = 1;

        while (a < A && b < B) {
            if (word == 1) {
                s.Add(word1[a].ToString());
                a += 1;
                word = 2;
            } else {
                s.Add(word2[b].ToString());
                b += 1;
                word = 1;
            }
        }
        while (a < A) {
            s.Add(word1[a].ToString());
            a += 1;
        }
        while (b < B) {
            s.Add(word2[b].ToString());
            b += 1;
        }

        return string.Join("", s);
    }
}